models/credmark/protocols/dexes/curve/curve_finance.py

```diff
@@ -2,23 +2,79 @@
 from credmark.cmf.model import Model
 from credmark.cmf.model.errors import ModelRunError
 from credmark.cmf.types.ledger import TransactionTable
-from credmark.dto import DTO
+
+from credmark.dto import (
+    DTO,
+    EmptyInput,
+)
+
 from credmark.cmf.types import (
     Address,
     Account,
     Accounts,
     Contract,
+    Contracts,
     Token,
-    Tokens
+    Tokens,
 )
-from credmark.cmf.types.series import BlockSeries
 
-from models.tmp_abi_lookup import (
-    CURVE_SWAP_ABI_1,
-    CURVE_SWAP_ABI_2,
-    CURVE_REGISTRY_ADDRESS,
-    CURVE_REGISTRY_ABI
-)
+
+@Model.describe(slug='curve-fi.get-provider',
+                version='1.1',
+                display_name='Curve Finance - Get Provider',
+                description='Get provider contract',
+                input=EmptyInput,
+                output=Contract)
+class CurveFinanceGetProvider(Model):
+    CURVE_PROVIDER_ALL_NETWORK = '0x0000000022D53366457F9d5E68Ec105046FC4383'
+
+    def run(self, _) -> Contract:
+        provider = Contract(address=Address(self.CURVE_PROVIDER_ALL_NETWORK).checksum)
+        return provider
+
+
+@Model.describe(slug='curve-fi.get-registry',
+                version='1.1',
+                display_name='Curve Finance - Get Registry',
+                description='Query provider to get the registry',
+                input=EmptyInput,
+                output=Contract)
+class CurveFinanceGetRegistry(Model):
+    def run(self, _) -> Contract:
+        provider = Contract(**self.context.models.curve_fi.get_provider())
+        reg_addr = provider.functions.get_registry().call()
+        return Contract(address=Address(reg_addr).checksum)
+
+
+@Model.describe(slug="curve-fi.get-gauge-controller",
```

Comment:
love it, let's think of a naming convention for this type of thing! the "go get the contract that has a list of contracts" type of thing